src/dbus_mpris.rs

```diff
@@ -698,6 +727,24 @@ async fn create_dbus_server(
     }
 }
 
+fn get_device_id(sp_client: &Arc<AuthCodeSpotify>, device_name: &String, only_active: bool) -> Option<Option<String>> {
+    let device_result = sp_client.device();
+    return match device_result {
+        Ok(devices) => devices.into_iter().find_map(|d| {
+            if d.name.eq(device_name) && (d.is_active || !only_active)  {
+                info!("Found device: {}, active: {}", d.name, d.is_active);
+                Some(d.id)
```

Comment:
I haven't dug into the code. Is there a reason, why we're returning a doubly wrapped `Option` instead of just `flatten`ing it?

Reply:
I don't see any, fixed.

Reply:
Oh, my bad. Instead of flattening it, it should probably just work by removing this `Some`.

Reply:
Oh, my bad. Instead of flattening it, it should probably just work by removing this `Some`.